### DISCUSSION TITLE
rocketmq水平扩容

### DISCUSSION THREAD

**Discussion by xhrg** [apache/rocketmq#8582] [General]
rocketmq做水平扩容，想把2主2从改为3主3从，那么需要在给新的broker创建topic和consumer，有没有办法一把梭快速创建。我自己测试了下，把topic.json文件复制过去是可以的，但是这个方案不知道是否合乎正常流程。

**Comment by cserwen** [comment]
目前没有拷贝 topic consumer 配置的工具，复制 topic.json 是比较直接有效的方法，受限于 rocketmq 没有中心化的元数据存储，目前只有这个策略

**Comment by asjfoajs** [comment]
可以用operator部署，在operator部署时，设置一个pod作为元数据，扩容时会自动复制这个pod的subscriptionGroup.json和topics.json然后写入到新扩容的pod里
